tokens/usecase/tokens_usecase.go

```diff
@@ -357,6 +363,64 @@ func (t *tokensUseCase) getChainScalingFactorMut(precision int) (osmomath.Dec, b
 	return result, ok
 }
 
+var (
+	jobIsRunning   bool
+	jobIsrunningMu sync.Mutex
+)
+
+// StartAsRoutineIfNotRunning starts the job in a new goroutine if it is not already running.
+func StartAsRoutineIfNotRunning(job func()) bool {
+	jobIsrunningMu.Lock()
+	if jobIsRunning {
+		jobIsrunningMu.Unlock()
+		return false
+	}
+	jobIsRunning = true
+	jobIsrunningMu.Unlock()
+
+	go func() {
+		jobIsrunningMu.Lock()
+		job()
+		jobIsRunning = false
+		jobIsrunningMu.Unlock()
+	}()
```

Comment:
Note that we don't really need this synchronization. We should be able to simply spin up a goroutine without any locks

Reply:
Implementation reflects the idea that only one assets update go routine can be running at any given time. For example if routine is already running second routine would be not launched to avoid DOS'ing external service.

I thought it can be useful for some edge cases:

- Wrong configuration of the update frequency ( for example every height )
- Network latency
 - Etc

Please let me know would you would like still to proceed to remove it and I will address that.

Reply:
I see. I would like to recommend considering using a worker pool [design pattern](https://shopify.engineering/leveraging-go-worker-pools) to address these edge cases.

We have the barebones abstractions defined [here](https://github.com/osmosis-labs/sqs/blob/v25.x/domain/workerpool/worker_pool.go)

By leveraging the abstraction, we will reuse the existing components, reducing complexity and addressing the raised concerns by configuring a pool with 1 concurrent worker.

However, I think that we can derisk them with a wider height interval that we can control with tests and observability.

I'll leave the worker pool decision up to you but I think that we should favor code simplicity here generally and derisk any concerns with tests and observability.

Reply:
For now I have removed to keep things simple as per you suggestion, we can always add this feature in the future if there is a need.